DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Utility.cs

```diff
@@ -0,0 +1,175 @@
+            int day = groups["day"].Success ? int.Parse(groups["day"].Value) : 1;
+            int hour = groups["hour"].Success ? int.Parse(groups["hour"].Value) : 0;
+            int minute = groups["minute"].Success ? int.Parse(groups["minute"].Value) : 0;
+            int second = groups["second"].Success ? int.Parse(groups["second"].Value) : 0;
+            int millisecond = groups["microsecond"].Success ? int.Parse(groups["microsecond"].Value) / 1000 : 0;
+
+            if (groups["timeZone"].Success)
+            {
+                int sign = int.Parse(groups["sign"].Value + "1");
+                int timeZoneHour = int.Parse(groups["timeZoneHour"].Value) * sign;
+                int timeZoneMinute = int.Parse(groups["timeZoneMinute"].Value) * sign;
+                return new DateTimeObject()
+                {
+                    DateValue = new DateTimeOffset(year, month, day, hour, minute, second, millisecond, new TimeSpan(timeZoneHour, timeZoneMinute, 0)),
+                    HasTimeZone = true,
+                };
+            }
+            else
+            {
+                return new DateTimeObject()
+                {
+                    DateValue = new DateTimeOffset(year, month, day, hour, minute, second, millisecond, default),
+                    HasTimeZone = false,
+                };
+            }
+        }
+
+        public static string GenerateDicomDateString(DateTimeOffset date)
+        {
+            return date.ToString("yyyyMMdd", CultureInfo.InvariantCulture);
+        }
+
+        public static string GenerateDicomDateTimeString(DateTimeObject date)
+        {
+            if (date == null)
+            {
+                return null;
+            }
+
+            return date.HasTimeZone == true
+                ? date.DateValue.ToString("yyyyMMddHHmmss.ffffffzzz", CultureInfo.InvariantCulture).Replace(":", string.Empty)
+                : date.DateValue.ToString("yyyyMMddhhmmss.ffffff", CultureInfo.InvariantCulture);
+        }
+
+        public static AgeValue ParseAge(string age)
+        {
+            if (string.IsNullOrEmpty(age))
+            {
+                return null;
+            }
+
+            foreach (var item in AgeTypeMapping)
+            {
+                if (new Regex(@"\d{3}" + item.Key).IsMatch(age))
+                {
+                    return new AgeValue(uint.Parse(age.Substring(0, AgeStringLength)), item.Value);
+                }
+            }
+
+            if (uint.TryParse(age, out uint result))
+            {
+                return new AgeValue(result, AgeType.Year);
+            }
+            else
+            {
+                throw new DicomDataException("Invalid age string. The valid strings are nnnD, nnnW, nnnM, nnnY.");
+            }
+        }
+
+        public static string AgeToString(AgeValue age)
+        {
+            if (age == null)
+            {
+                return null;
```

Comment:
>null [](http://example.com/codeflow?start=23&length=4)

just curious, can the caller of this function receive a null value or a better return value is EmptyString?

Reply:
caller will ignore the result when return null.